docs/build_docs.py

```diff
@@ -0,0 +1,898 @@
+                "Using the environmental variable $GEOIPS_REPO_URL as value for "
+                "--license-url. This functionality is deprecated and will be removed. "
+                "Please pass in using --license-url $GEOIPS_REPO_URL. Alternatively,"
+                "you can unset $GEOIPS_REPO_URL and the URL will default to"
+                "https://github.com/NRLMDD-GEOIPS/package_name"
+            )
+        else:
+            repo_url = "https://github.com/NRLMMD-GEOIPS/" + args.package_name
+        args.license_url = repo_url
+
+    return args
+
+
+def validate_path_exists_and_is_git_repo(repo_path, logger=logging.getLogger(__name__)):
+    """
+    Ensure given path exists and is a valid Git repository.
+
+    Parameters
+    ----------
+    repo_path : str
+        The file system path to validate.
+    logger : logging.Logger, optional
+        Logger for logging messages; defaults to the module logger.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the provided path does not exist.
+    pygit2.GitError
+        If the path is not a valid Git repository.
+    """
+    if not os.path.exists(repo_path):
+        logger.critical(f"Path does not exist: {repo_path}")
+        raise FileNotFoundError
+    try:
+        pygit2.Repository(repo_path)
+    except pygit2.GitError as e:
+        logger.critical(f"{repo_path} is not a valid git repo")
+        raise e
+
+
+def validate_package_is_installed(package_name, logger=logging.getLogger(__name__)):
+    """
+    Verify that the specified package is installed in the current Python environment.
+
+    Parameters
+    ----------
+    package_name : str
+        The name of the package to check for installation.
+    logger : logging.Logger, optional
+        Logger for logging messages; defaults to the module logger.
+
+    Raises
+    ------
+    ModuleNotFoundError
+        If the package is not found in the current Python environment.
+    """
+    if importlib.util.find_spec(package_name) is None:
+        logger.critical(f"ERROR: Package {package_name} is not installed")
+        raise ModuleNotFoundError
+
+
+def get_section_replace_string(section):
+    """
+    Create a placeholder string used in the index template for a given section.
+
+    Parameters
+    ----------
+    section : str
+        The name of the section.
+
+    Returns
+    -------
+    str
+        A placeholder string corresponding to the section.
+
+    Notes
+    -----
+    It is unclear WHY this is the placeholder string or what IDX stands for.
+    """
+    return section.upper() + "IDX"
+
+
+def get_sections():
+    """
+    Return sections.
+
+    Returns a list of required and optional sections.
+
+    Returns
+    -------
+    required_sections : list of str
+        List of required section names.
+    optional_sections : list of str
+        List of optional section names.
+
+    Notes
+    -----
+    This could potentially be from a config file in the future. For now, it is hard
```

Comment:
Will this same doc build be used with the new docs, just update this section / make it a config?

Reply:
Yes! Same doc build with some tweaks for the new docs :)